x/paychan/handler.go

```diff
@@ -69,9 +69,11 @@ func (h *createPaymentChannelHandler) Deliver(ctx weave.Context, db weave.KVStor
 		return nil, err
 	}
 
-	nextID, err := peekNextID(db)
+	// Explicitely acquire the ID as we need it before saving to compute
```

Comment:
Now this is actually a cool idea. Just pre-generating the id and using it for `Put`.